Server/htdocs/AppController/commands_RSM/utilities/RStools.php

```diff
@@ -539,4 +539,97 @@ function is_base64($s){
 
     return true;
 }
-?>
+
+// Set the Authorization token readed on the header and puts it in the $GLOBALS variable 
+function setAuthorizationTokenOnGlobals() {
+    //We need this variable to exists in order to make RSdatabase work propertly.
+    $GLOBALS['RS_POST']['RStoken'] = getallheaders()["Authorization"];
+    if (strpos($GLOBALS['RS_POST']['RStoken'], "Basic ") !== false) {
+        $GLOBALS['RS_POST']['RStoken'] = str_replace("Basic ", "", $GLOBALS['RS_POST']['RStoken']);
+        $GLOBALS['RS_POST']['RStoken'] =  base64_decode($GLOBALS['RS_POST']['RStoken']);
+        $GLOBALS['RS_POST']['RStoken'] = str_replace("Authorization:", "", $GLOBALS['RS_POST']['RStoken']);
+        $GLOBALS['RS_POST']['RStoken'] = str_replace(";", "", $GLOBALS['RS_POST']['RStoken']);
+    }
+}
+
+// Replace incorrect values related to UTF-8
+function replaceUtf8Characters($propertyValue) {
+    $parsedPropertyValue = str_replace("&amp;", "&", htmlentities($propertyValue, ENT_COMPAT, "UTF-8"));
+    return str_replace("'", "&#39;", $parsedPropertyValue);
+}
+
+// Returns request body sent through petition, transormed into php object (json)
+function getRequestBody() {
+    global $RSallowDebug;
+    $body = json_decode(stripslashes(file_get_contents('php://input')));
+    if ($body == "") {
+        if ($RSallowDebug) returnJsonMessage(400, "Invalid JSON body");
+        else returnJsonMessage(400, "");
```

Comment:
In all this file, if $RSallowDebug is false, there won't be tracer in the PHP error log. If the error is not displayed in the screen, it should be displayed in the log.

Reply:
This happens in all functions

Reply:
Lo cambiamos, y aprovecharemos para crear una función común para devolver los JsonMessage y los errores de consola (así evitaremos tener if/else en todas partes).